src/Microsoft.ML.Core/Data/IHostEnvironment.cs

```diff
@@ -63,14 +63,23 @@ public interface IHostEnvironment : IChannelProvider, IProgressChannelProvider
         IHost Register(string name, int? seed = null, bool? verbose = null);
 
         /// <summary>
-        /// Flag which indicate should we stop any code execution in this host.
+        /// The catalog of loadable components (<see cref="LoadableClassAttribute"/>) that are available in this host.
         /// </summary>
-        bool IsCancelled { get; }
+        ComponentCatalog ComponentCatalog { get; }
+    }
 
+    [BestFriend]
+    internal interface ICancelableHost
+    {
         /// <summary>
-        /// The catalog of loadable components (<see cref="LoadableClassAttribute"/>) that are available in this host.
+        /// Signal to stop exection in this host and all its children.
```

Comment:
>host and all its children [](start = 44, length = 25)

but will it cancel parents, siblings and environment?

Reply:
Hmm...I added a test that verifies this. Did you get a chance to look at it?